src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -304,6 +320,14 @@ public override string Name
             }
         }
 
+        public override string MetadataName
+        {
+            get
+            {   
+                return HasNameInMetadata ? _name : string.Empty;
```

Comment:
Yes, it is possible to get confused with "null" terminology. I got confused myself. Index 0 is in fact what ECMA spec calls null, and it turns out, when we import/emit strings values from/to metadata, it is equivalent to an empty string.